### PR TITLE
hwdb,udevadm: unify duplicated code and move it to hwdb-util.c

### PR DESCRIPTION
**Pull request by yuwata**
udevadm-hwdb was deprecated more than three years ago. Let's unify the code in systemd-hwdb.c and udevadm-hwdb.c

This also adds `--compat` option to `systemd-hwdb`. With the option, `systemd-hwdb` generates hwdb.bin which compatible with `udevadm hwdb`.

### CODE REVIEW

**Comment by yuwata** [comment]
Updated. PTAL.

File: src/libsystemd/sd-hwdb/hwdb-util.c [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+        const struct trie_value_entry *val2 = v2;
+        struct trie *trie = arg;
+
+        return strcmp(trie->strings->buf + val1->key_off,
+                      trie->strings->buf + val2->key_off);
+}
+
+static int trie_node_add_value(struct trie *trie, struct trie_node *node,
+                               const char *key, const char *value,
+                               const char *filename, uint16_t file_priority, uint32_t line_number, bool compat) {
```
Comment (keszybz) [inline src/libsystemd/sd-hwdb/hwdb-util.c:135]: I'd appreciate a description in the commit header of what `compat` means and in general what the situation with compatibility is.

File: src/hwdb/hwdb.c [hunk 2]
```diff
@@ -61,6 +62,7 @@ static int help(void) {
                "     --version    Show package version\n"
                "  -s --strict     When updating, return non-zero exit value on any parsing error\n"
                "     --usr        Generate in " UDEVLIBEXECDIR " instead of /etc/udev\n"
+               "     --compat     Generate database in old format\n"
```
Comment (keszybz) [inline src/hwdb/hwdb.c]: `s/old format/old format read by udevadm/` ?

<strike>This also needs a man page entry, with some guidance when the `--compat` format should be used.</strike>

**Comment by yuwata** [comment]
There is a good explanation here: https://github.com/systemd/systemd/blob/124b3686f97a022c7c04e25edb06ef547492c497/src/libsystemd/sd-hwdb/sd-hwdb.c#L147-L169

I've not tested, but I guess that old udev uses the old format, but it _can_ read the new format. Moreover, 'udevadm hwdb --update' creates database in old format. So, ok, it seems not necessary to add '--compat' option, at least until udevadm 'supports' hwdb command.

File: src/libsystemd/sd-hwdb/hwdb-util.c (hunk 1 above)
Comment (yuwata) [reply src/libsystemd/sd-hwdb/hwdb-util.c:135]: OK.

**Comment by yuwata** [comment]
@keszybz Updated.
- The new option '--compat' is dropped.
- Adds some comments in the source and commit message about the argument 'compat' for 'hwdb_update()'.
- Also moves the default paths to hwdb-util.c.

**Comment by keszybz** [comment]
Looks nice. I think we can probably make `udevadm hwdb` also generate the new format and drop the compat code, but this can be done in a separate PR. It's nice that this PR is only about refactoring.

**Pull request merged**